### PR TITLE
[risk=no]source tree bug fix

### PR DESCRIPTION
**Pull request by mhigbyflowers**

### CODE REVIEW

**Review by SrushtiGangireddy — APPROVED** [review]
Looks good. thanks.

File: public-ui/src/app/components/recursive-tree/recursive-tree.component.ts [hunk 1]
```diff
@@ -9,8 +9,10 @@ import { DataBrowserService } from '../../../publicGenerated/api/dataBrowser.ser
 })
 export class RecursiveTreeComponent implements OnChanges, OnDestroy {
   @Input() node: any;
-  opened = false;
+  @Input() opened: boolean;
```
Comment (mhigbyflowers) [inline public-ui/src/app/components/recursive-tree/recursive-tree.component.ts:12]: passed opened to from the parent and then to itself

File: public-ui/src/app/components/recursive-tree/recursive-tree.component.ts [hunk 2]
```diff
@@ -9,8 +9,10 @@ import { DataBrowserService } from '../../../publicGenerated/api/dataBrowser.ser
...
+  @Input() childIndex: number;
```
Comment (mhigbyflowers) [inline public-ui/src/app/components/recursive-tree/recursive-tree.component.ts:13]: Sets its index for specifically.

File: public-ui/src/app/components/recursive-tree/recursive-tree.component.ts [hunk 3]
```diff
@@ -25,13 +27,21 @@ export class RecursiveTreeComponent implements OnChanges, OnDestroy {
...
+      );
+    }
   }
   ngOnDestroy() {
     for (const s of this.subscriptions) {
       s.unsubscribe();
     }
   }
-
+  openUp(index, node, nodes) {
```
Comment (mhigbyflowers) [inline public-ui/src/app/components/recursive-tree/recursive-tree.component.ts:38]: sets the index.

**Pull request merged**